homeassistant/components/sensor/__init__.py

```diff
@@ -510,65 +567,91 @@ def state(self) -> Any:
 
         # If the sensor has neither a device class, a state class nor
         # a unit_of measurement then there are no further checks or conversions
-        if not device_class and not state_class and not unit_of_measurement:
+        if (
+            not device_class
+            and not state_class
+            and not unit_of_measurement
+            and (prec := self.precision) is None
+        ):
             return value
 
-        if not self._invalid_numeric_value_reported and not isinstance(
-            value, (int, float, Decimal)
-        ):
+        numerical_value: int | float | Decimal | None = None
```

Comment:
Let's add a comment and remove None
```suggestion
        # From now only numeric values are valid
        numerical_value: int | float | Decimal
```